File: Crio/crio_programming_interview_problems-master/MaximumSubArray/MaximumSubArray.java
import java.io.*;
import java.util.*;

public class MaximumSubArray {
    // Implement your solution by completing the below function
    public int maxSubArray(int[] nums) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = nums.length;
        int currSum = nums[0], maxSum = nums[0];

        for (int i = 1; i < n; ++i) {
            currSum = Math.max(nums[i], currSum + nums[i]);
            maxSum = Math.max(maxSum, currSum);
        }
        return maxSum;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

        // CRIO_UNCOMMENT_START_MODULE_L1_PROBLEMS
        //return Integer.MIN_VALUE;
        // CRIO_UNCOMMENT_END_MODULE_L1_PROBLEMS
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int[] nums = new int[scanner.nextInt()];
        for (int i = 0; i < nums.length; i++)
            nums[i] = scanner.nextInt();
        scanner.close();

        int result = new MaximumSubArray().maxSubArray(nums);
        System.out.print(Integer.toString(result));
    }
}

File: Crio/crio_programming_interview_problems-master/MaximumSubArray/MaximumSubArray.cpp
#include <bits/stdc++.h>
#include "FastIO.hpp"
#include "ReadMatrix.hpp"

using namespace std;

class MaximumSubArray {
    public:
        int maxSubArray(vector<int>& nums) {
            int ans = 0;
            // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
            int n=nums.size();
            if(n==0) return 0;
            if(n==1) return  nums[0];
            int dp[n];
            dp[0]=nums[0];
            ans=dp[0];
            for(int i=1;i<n;i++)
            {
                int x=dp[i-1]+nums[i];
                dp[i]=max(nums[i],x);
                ans=max(ans,dp[i]);
            }
            // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
            return ans;
        }
};

int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    int result = MaximumSubArray().maxSubArray(nums);
    cout << result;
    return 0;
}
